src/parser.rs

```diff
@@ -57,15 +57,15 @@ pub enum ParseLocation {
 #[derive(PartialEq, Eq)]
 pub struct ParseError {
     kind: ParseErrorKind,
-    parse_locations: [Option<ParseLocation>; 4],
+    parse_locations: Box<[Option<ParseLocation>; 4]>,
```

Comment:
I'm not wild about Box here. We're currently alloc free in parsing.

Reply:
Makes sense -- I think the only other options are to either reduce `parse_locations` to 3 contexts (making debugging/identifying the location harder?) _or_ disable this clippy lint.

Reply:
Or we could reject @reaperhulk's suggestion and go "actual only" :D

Reply:
Disabled the `result_large_err` lint outright. Another option here might be to keep it, but to set `large-error-threshold` to the current limit to prevent further accidental growth: <https://doc.rust-lang.org/nightly/clippy/lint_configuration.html#large-error-threshold>

Reply:
That would definitely be better

Reply:
Done!